code/parse/sexp.cpp

```diff
@@ -2636,13 +2636,20 @@ int check_sexp_syntax(int node, int return_type, int recursive, int *bad_node, s
 						if (get_operator_const(op_node) < First_available_operator_id) {
 							ship_node = CDR(op_node);
 						} else {
-							ship_node = get_dynamic_parameter_index(Sexp_nodes[op_node].text, argnum);
-
-							if (ship_node < 0)
-								error_display(1, "Expected to find a dynamic lua parent parameter for node %i in operator %s but found nothing!",
+							int r_count = get_dynamic_parameter_index(Sexp_nodes[op_node].text, argnum) + 1; //account for the node header
+							
+							if (r_count < 0)
+								error_display(1,
+									"Expected to find a dynamic lua parent parameter for node %i in operator %s but "
+									"found nothing!",
 									argnum,
 									Sexp_nodes[op_node].text);
-
+							
+							ship_node = 0; //initialize it I guess
+							while (r_count >= 0) {
+								ship_node = CDR(op_node);
+								r_count--;
+							}
```

Comment:
This won't iterate over the nodes correctly.  CDR moves to the node's first child, so this will just take the first child, then the first child again, etc.  You'll need to do `ship_node = CDR(ship_node)`.  Consequently you can initialize `ship_node` to `op_node`.

Reply:
Done